python/tests/test_dict_encoding.py

```diff
@@ -21,595 +21,16 @@
-            "mutations",
-            ["site", "node", "derived_state", "derived_state_offset"],
-        )
-        self.verify_offset_pair(tables, len(tables.mutations), "mutations", "metadata")
-        self.verify_metadata_schema(tables, "mutations")
-        # Verify optional time column
-        d = tables.asdict()
-        d["mutations"]["time"] = None
-        lwt = c_module.LightweightTableCollection()
-        lwt.fromdict(d)
-        out = lwt.asdict()
-        self.assertTrue(
-            all(util.is_unknown_time(val) for val in out["mutations"]["time"])
-        )
-
-    def test_populations(self):
-        tables = get_example_tables()
-        self.verify_required_columns(
-            tables, "populations", ["metadata", "metadata_offset"]
-        )
-        self.verify_metadata_schema(tables, "populations")
-        self.verify_offset_pair(tables, len(tables.nodes), "nodes", "metadata", True)
-
-    def test_provenances(self):
-        tables = get_example_tables()
-        self.verify_required_columns(
-            tables,
-            "provenances",
-            ["record", "record_offset", "timestamp", "timestamp_offset"],
-        )
-
-    def test_top_level_metadata(self):
-        tables = get_example_tables()
-        d = tables.asdict()
-        # None should give default value
-        d["metadata"] = None
-        lwt = c_module.LightweightTableCollection()
-        lwt.fromdict(d)
-        out = lwt.asdict()
-        self.assertNotIn("metadata", out)
-        tables = tskit.TableCollection.fromdict(out)
-        # We only removed the metadata, not the schema. So empty bytefield
-        # still gets interpreted
-        self.assertEqual(tables.metadata, {"top-level": []})
-        # Missing is tested in TestMissingData above
-
-    def test_top_level_metadata_schema(self):
-        tables = get_example_tables()
-        d = tables.asdict()
-        # None should give default value
-        d["metadata_schema"] = None
-        lwt = c_module.LightweightTableCollection()
-        lwt.fromdict(d)
-        out = lwt.asdict()
-        self.assertNotIn("metadata_schema", out)
-        tables = tskit.TableCollection.fromdict(out)
-        self.assertEqual(str(tables.metadata_schema), "")
-        # Missing is tested in TestMissingData above
-
-
-class TestExamples(unittest.TestCase):
-    def test_pickled_examples(self):
-        seen_msprime = False
-        test_dir = pathlib.Path(__file__).parent / "data/dict-encodings"
-        for filename in test_dir.glob("*.pkl"):
-            if "msprime" in str(filename):
-                seen_msprime = True
-            with open(test_dir / filename, "rb") as f:
-                d = pickle.load(f)
-                lwt = c_module.LightweightTableCollection()
-                lwt.fromdict(d)
-                tskit.TableCollection.fromdict(d)
-        # Check we've done something
-        self.assertTrue(seen_msprime)
+from tskit.dict_encoding_tests import *  # noqa: F403,F401
```

Comment:
It's a good idea, but this will mean that we use the version of the tests from the *installed* version of tskit, not the version which corresponds to the C-lib version. These could get out of date - suppose we have msprime using version 0.99.X and everything is working fine. Then we release a tskit version which is based on 0.99.X + 1, which has an added some extra fields to the data model that are irrelevant for msprime. Then, these tests will break. It would essentially force us to update the C API in all downstream packages everytime we update the C API in Python tskit, which I don't think we want to do.

Reply:
Probably simplest to mess around with this downstream and see what can be made work. Maybe we merge the basic change first, and see how we go? It'll make things easier to test if we can point msprime at a hash in the main repo rather than pointing at a fork.

Reply:
Yes, the tskit-using-application tests need to pull from the tests from the submodule of tskit, not the installed tskit. So it's either a weird looking import or a symlink I think?

Reply:
Ok, lets back out the last commit i did, then merge.